service/internal/servicetelemetry/telemetry_settings.go

```diff
@@ -43,12 +43,17 @@ type TelemetrySettings struct {
 // ToComponentTelemetrySettings returns a TelemetrySettings for a specific component derived from
 // this service level Settings object.
 func (s TelemetrySettings) ToComponentTelemetrySettings(id *component.InstanceID) component.TelemetrySettings {
+	status := status.NewComponentStatusFunc(id, s.Status.ReportStatus)
```

Comment:
Super minor nit, can we call this `statusFunc`?

Reply:
Sure!

Reply:
done